src/test/java/com/exasol/spark/s3/S3IntegrationTestSetup.java

```diff
@@ -26,27 +28,51 @@ public abstract class S3IntegrationTestSetup extends BaseIntegrationSetup {
     protected static final String DEFAULT_BUCKET_NAME = "csvtest";
 
     @Container
-    private static final LocalStackContainer S3 = new LocalStackContainer(
-            DockerImageName.parse("localstack/localstack:2.0.2")).withServices(Service.S3);
+    protected static final LocalStackContainer S3 = new LocalStackContainer(
+            DockerImageName.parse("localstack/localstack:2.0")) //
+            .withServices(Service.S3) //
+            .withReuse(true);
 
-    // protected static S3Client s3Client;
+    protected static S3Client s3Client;
 
     @BeforeAll
     public static void setup() throws SQLException {
         LOGGER.info(() -> "Created localstack S3 client with region '" + S3.getRegion() + "'.");
-        // s3Client = S3Client.builder() //
-        //         .endpointOverride(S3.getEndpointOverride(Service.S3)) //
-        //         .credentialsProvider(StaticCredentialsProvider
-        //                 .create(AwsBasicCredentials.create(S3.getAccessKey(), S3.getSecretKey()))) //
-        //         .region(Region.of(S3.getRegion())) //
-        //         .build();
-        // updateExasolContainerHostsFile();
+        s3Client = S3Client.builder() //
+                .endpointOverride(S3.getEndpointOverride(Service.S3)) //
+                .credentialsProvider(StaticCredentialsProvider
+                        .create(AwsBasicCredentials.create(S3.getAccessKey(), S3.getSecretKey()))) //
+                .region(Region.of(S3.getRegion())) //
+                .build();
+        updateHostsFileInExasol();
         createBucket(DEFAULT_BUCKET_NAME);
     }
 
     public static void createBucket(final String bucketName) {
         LOGGER.info(() -> "Creating S3 bucket '" + bucketName + "'.");
-        // s3Client.createBucket(b -> b.bucket(bucketName));
+        s3Client.createBucket(b -> b.bucket(bucketName));
+    }
+
+    private static void updateHostsFileInExasol() {
+        final List<String> commands = Arrays.asList( //
+                "sed -i '/amazonaws/d' /etc/hosts", //
+                "echo '" + getS3ContainerInternalIp() + " csvtest.s3.amazonaws.com' >> /etc/hosts" //
+        );
+        commands.forEach(command -> {
+            try {
+                final ExecResult exitCode = EXASOL.execInContainer("/bin/sh", "-c", command);
+                if (exitCode.getExitCode() != 0) {
+                    throw new RuntimeException(
+                            "Command to update Exasol container `/etc/hosts` file returned non-zero result.");
+                }
+            } catch (final InterruptedException | IOException exception) {
+                throw new RuntimeException("Failed to update Exasol container `/etc/hosts`.", exception);
+            }
+        });
+    }
```

Comment:
I propose to use a constant for the file path and to use a method name that indicates the intention (redirect ip address) rather than how to achieve that (modify hosts file).
```suggestion
    private static void redirectIpAddress(final ExasolContainer<?> exasolContainer, final String original,
            final String redirect) {
        final List<String> commands = Arrays.asList( //
                "sed -i '/amazonaws/d' " + HOSTS_FILE, // maybe use original here?
                "echo '" + redirect + " " + original + "' >> " + HOSTS_FILE);
        commands.forEach(command -> {
            try {
                final ExecResult exitCode = exasolContainer.execInContainer("/bin/sh", "-c", command);
                if (exitCode.getExitCode() != 0) {
                    throw new RuntimeException(
                            "Command to update Exasol container '" + HOSTS_FILE + "' file returned non-zero result.");
                }
            } catch (final InterruptedException | IOException exception) {
                throw new RuntimeException("Failed to update Exasol container '" + HOSTS_FILE + "'.", exception);
            }
        });
    }
```